libs/base/core.cpp

```diff
@@ -561,49 +561,50 @@ int length(String s) {
 }
 
 #define isspace(c) ((c) == ' ')
+#define iswhitespace(c) ((c) == 0x0009 || (c) == 0x000B || (c) == 0x000C || (c) == 0x0020 || (c) == 0x00A0 || (c) == 0xFEFF || (c) == 0x000A || (c) == 0x000D || (c) == 0x2028 || (c) == 0x2029)
 
 NUMBER mystrtod(const char *p, char **endp) {
-    while (isspace(*p))
+    while (iswhitespace(*p))
         p++;
     NUMBER m = 1;
     NUMBER v = 0;
     int dot = 0;
+    int hasDigit = 0;
     if (*p == '+')
         p++;
     if (*p == '-') {
         m = -1;
         p++;
     }
-    if (*p == '0' && (p[1] | 0x20) == 'x') {
-        return m * strtol(p, endp, 16);
-    }
+
     while (*p) {
         int c = *p - '0';
         if (0 <= c && c <= 9) {
             v *= 10;
             v += c;
             if (dot)
                 m /= 10;
+            hasDigit = 1;
         } else if (!dot && *p == '.') {
             dot = 1;
-        } else if (*p == 'e' || *p == 'E') {
-            break;
+        } else if (!hasDigit) {
+            return NAN;
         } else {
-            while (isspace(*p))
-                p++;
-            if (*p)
-                return NAN;
             break;
         }
         p++;
     }
 
     v *= m;
 
-    if (*p) {
+    if (*p == 'e' || *p == 'E') {
         p++;
         int pw = (int)strtol(p, endp, 10);
-        v *= p10(pw);
+        if (!isnan(pw)) {
```

Comment:
int can never be nan, it will return 0 if no number is found, and keep `*endp == p`; in other words, I think you can skip the entire if statement